File: UI/src/app/shared/helpers/date.helper.ts
import { Period } from "src/app/domain/period/period.dto";

export class DateHelper {
	public static FromPeriod(period: Period) {
		const today = new Date();

		today.setMonth(period.month - 1);
		today.setFullYear(period.year);

		return today
	}

	public static PeriodNow(): Period {
		const month = new Date().getMonth() + 1;
		const year = new Date().getFullYear();

		return { month, year };
	}
}

File: WebAPI/Core/Helpers/DateHelper.cs
using Financial.Domain.DTO;

namespace Financial.Core.Helpers
{
    public static class DateHelper
    {
        public static bool InTimeRange(DateTimeOffset DateToCheck, DateTimeOffset LowerBound, DateTimeOffset UpperBound)
        {
            return (DateToCheck.Year == LowerBound.Year && DateToCheck.Month >= LowerBound.Month || DateToCheck.Year > LowerBound.Year)
                    && (DateToCheck.Year == UpperBound.Year && DateToCheck.Month <= UpperBound.Month || DateToCheck.Year < UpperBound.Year);
        }

        public static bool InTimeRange(Period DateToCheck, DateTimeOffset LowerBound, DateTimeOffset UpperBound)
        {
            return (DateToCheck.Year == LowerBound.Year && DateToCheck.Month >= LowerBound.Month || DateToCheck.Year > LowerBound.Year)
                    && (DateToCheck.Year == UpperBound.Year && DateToCheck.Month <= UpperBound.Month || DateToCheck.Year < UpperBound.Year);
        }
    }
}
